src/main/ml-modules/ext/com.marklogic.smart-mastering/matcher-impl/matcher-impl.xqy

```diff
@@ -72,23 +72,26 @@ declare function match-impl:find-document-matches-by-options(
     if ($lock-on-search) then
       match-impl:lock-on-search($serialized-match-query/cts:and-query/cts:or-query)
     else ()
+  let $results :=
+    match-impl:search(
+      $match-query,
+      $reduced-boost,
+      $minimum-threshold,
+      $thresholds,
+      $start,
+      $page-length,
+      $scoring,
+      $algorithms,
+      $options,
+      $include-matches
+    )
   return (
     $_lock-on-search,
     element results {
+      attribute total { fn:count($results) },
```

Comment:
`fn:count($results)` is subject to `$start` and `$page-length`. `@total` should be based on an `xdmp:estimate` using the query.

Reply:
doh!